src/image.c

```diff
@@ -31,27 +31,55 @@ typedef enum {
 #undef _
 } twin_image_format_t;
 
-/* FIXME: Check the header of the given images to determine the supported image
- * formats instead of parsing the filename without checking its content.
+/*
+ * Defines the headers of supported image formats.
+ * Each image format has a unique header, allowing the format to be
+ * determined by inspecting the file header.
+ * Supported formats: PNG, JPEG.
+ * Reference links:
+ * - PNG:
+ * http://www.libpng.org/pub/png/spec/1.2/PNG-Rationale.html#R.PNG-file-signature
+ * - JPEG:
+ * https://www.file-recovery.com/jpg-signature-format.htm
  */
+#if __BYTE_ORDER == __BIG_ENDIAN
+static const uint8_t header_png[8] = {
+    0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A,
+};
+#else
+static const uint8_t header_png[8] = {
+    0x0A, 0x1A, 0x0A, 0x0D, 0x47, 0x4E, 0x50, 0x89,
+};
+#endif
+static const uint8_t header_jpeg[3] = {0xFF, 0xD8, 0xFF};
+
 static twin_image_format_t image_type_from_name(const char *path)
 {
     twin_image_format_t type = IMAGE_TYPE_unknown;
-    const char *extname = strrchr(path, '.');
-    if (!extname)
+    FILE *file = fopen(path, "rb");
+    if (!file) {
+        perror("Failed to open %s");
```

Comment:
This format string is malformed without a valid C-style string.

Reply:
Based on my understanding, maybe replace `perror` with 
`
fprintf(stderr, "Failed to open %s\n", path);
`

Reply:
It is not necessarily polite to use phrases such as "based on my understanding" or "to my knowledge." Instead, you should emphasize facts and propose specific code changes during collaborative conversations.